### ISSUE TITLE
Accessibility. Crash on Windows when traversing elements

### ISSUE THREAD

**Issue by igordmn** [JetBrains/compose-multiplatform#4778]
Compose 1.6.10-rc01, Windows 11, NVDA 2023.2

1. run run1
2. traverse with hot keys
3. see crash:
```
Caused by: java.lang.IndexOutOfBoundsException: index: -3, size: 1
	at kotlin.collections.AbstractList$Companion.checkElementIndex$kotlin_stdlib(AbstractList.kt:108)
	at kotlin.collections.builders.ListBuilder.get(ListBuilder.kt:59)
	at androidx.compose.ui.platform.a11y.ComposeAccessible$ComposeAccessibleComponent.getAccessibleChild(ComposeAccessible.kt:337)
	at androidx.compose.ui.platform.a11y.ComposeAccessible$ComposeAccessibleComponent.getAccessibleAt(ComposeAccessible.kt:380)
	at androidx.compose.ui.platform.a11y.ComposeAccessible$ComposeAccessibleComponent.getAccessibleAt(ComposeAccessible.kt:381)
	at androidx.compose.ui.platform.a11y.ComposeAccessible$ComposeAccessibleComponent.getAccessibleAt(ComposeAccessible.kt:381)
	at androidx.compose.ui.platform.a11y.ComposeSceneAccessible$ComposeSceneAccessibleContext.getAccessibleAt(ComposeSceneAccessible.kt:89)
	at jdk.accessibility/com.sun.java.accessibility.util.EventQueueMonitor.getAccessibleAt(EventQueueMonitor.java:441)
	at jdk.accessibility/com.sun.java.accessibility.internal.AccessBridge$12.call(AccessBridge.java:581)
	at jdk.accessibility/com.sun.java.accessibility.internal.AccessBridge$12.call(AccessBridge.java:578)
	at jdk.accessibility/com.sun.java.accessibility.internal.AccessBridge$InvocationUtils$CallableWrapper.run(AccessBridge.java:7334)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:308)
```

**Comment by m-sasha** [comment]
I'll take a look soon.

**Comment by m-sasha** [comment]
It reproduces easily on macOS with Compose 1.6.10, but does not reproduce in `jb-main` tip or in `1.7.0-dev1686`.

@igordmn Can you check on Windows?
@mgroth0 Can you check with Compose `1.7.0-dev1686`?

**Comment by m-sasha** [comment]
For me it reproduces by pressing TAB after scrolling the list down with the scroll wheel.

**Comment by igordmn** [comment]
Reproduced on Windows,`1.7.0-dev1686`/`1.6.10`, just scrolling up/down.

After closing NVDA - it doesn't reproduce (probably because accessibility functions aren't called)

**Comment by m-sasha** [comment]
This should be fixed now. The latest reproducer you posted does not reproduce after the fix.

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-4778">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
